src/models/Hex.ts

```diff
@@ -1,20 +1,36 @@
-export class Hex {
-  public q: number
-  public r: number
-  public s: number
+export interface HexCoordinates {
+  q: number
+  r: number
+  s: number
+}
+type HexPropsType = {
+  fill: string
+  className: string
+}
+interface HexAttributes {
+  blocked: boolean
+  text: string
+  image: string
+  props: Partial<HexPropsType>
+  state: any
+  pattern: string
+}
+export class Hex implements HexCoordinates, Partial<HexAttributes> {
```

Comment:
I thinks this whole class is not really usefull. Your changes seem to be sensible though.

Reply:
Now that we're using TypeScript, we can actually change Hex from a class to an interface/type since it's not really doing anything except acting as a type. Doing that would possibly introduce a ton of breaking changes so I'll create another branch for that later. It's probably easy enough.